grype/pkg/package_test.go

```diff
@@ -0,0 +1,229 @@
+							"named-section": {
+								"named-section-key": "named-section-vaulue",
+							},
+						},
+					},
+					PomProperties: &syftPkg.PomProperties{
+						Path:       "pom-path-info",
+						Name:       "pom-name-info",
+						GroupID:    "pom-group-id-info",
+						ArtifactID: "pom-artifact-id-info",
+						Version:    "pom-version-info",
+						Extra: map[string]string{
+							"extra-key": "extra-value",
+						},
+					},
+				},
+			},
+			metadata: JavaMetadata{
+				VirtualPath:   "virtual-path-info",
+				PomArtifactID: "pom-artifact-id-info",
+				PomGroupID:    "pom-group-id-info",
+				ManifestName:  "main-section-name-info",
+			},
+		},
+		// below cases we assert that there is no mapped metadata
+		{
+			name: "apk-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.ApkMetadataType,
+				Metadata: syftPkg.ApkMetadata{
+					Package:          "a",
+					OriginPackage:    "a",
+					Maintainer:       "a",
+					Version:          "a",
+					License:          "a",
+					Architecture:     "a",
+					URL:              "a",
+					Description:      "a",
+					Size:             1,
+					InstalledSize:    1,
+					PullDependencies: "a",
+					PullChecksum:     "a",
+					GitCommitOfAport: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "npm-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.NpmPackageJSONMetadataType,
+				Metadata: syftPkg.NpmPackageJSONMetadata{
+					Author:      "a",
+					Homepage:    "a",
+					Description: "a",
+					URL:         "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "python-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.PythonPackageMetadataType,
+				Metadata: syftPkg.PythonPackageMetadata{
+					Name:                 "a",
+					Version:              "a",
+					License:              "a",
+					Author:               "a",
+					AuthorEmail:          "a",
+					Platform:             "a",
+					SitePackagesRootPath: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "gem-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.GemMetadataType,
+				Metadata: syftPkg.GemMetadata{
+					Name:     "a",
+					Version:  "a",
+					Homepage: "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "kb-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.KbPackageMetadataType,
+				Metadata: syftPkg.KbPackageMetadata{
+					ProductID: "a",
+					Kb:        "a",
+				},
+			},
+			metadata: nil,
+		},
+		{
+			name: "rust-metadata",
+			syftPkg: syftPkg.Package{
+				MetadataType: syftPkg.RustCargoPackageMetadataType,
+				Metadata: syftPkg.CargoPackageMetadata{
+					Name:     "a",
+					Version:  "a",
+					Source:   "a",
+					Checksum: "a",
+				},
+			},
+			metadata: nil,
+		},
+	}
+
+	// capture each observed metadata type, we should see all of them relate to what syft provides by the end of testing
+	observedMetadataTypes := set.NewStringSet()
```

Comment:
nit: I'd declare `observedMetadataTypes` down in the stanza that starts to populate the set (same scope, just repositioned down a few lines). I don't feel strongly about this, but it helped me mentally to picture setting up "expected things" first, and then starting to observe.